Mathlib/Combinatorics/SimpleGraph/Walk.lean

```diff
@@ -809,6 +809,23 @@ theorem nodup_tail_support_reverse {u : V} {p : G.Walk u u} :
     ← getVert_eq_support_getElem? _ (by rw [Walk.length_support]; cutsat)]
   aesop
 
+theorem edges_eq_support {u v : V} {p : G.Walk u v} :
+    p.edges = (p.support.zip p.support.tail).map (fun x ↦ s(x.1, x.2)) := by
+  induction p with
+  | nil => simp
+  | cons _ p' ih => cases p' <;> simp [edges_cons, ih]
+
+theorem darts_toProd_eq {u v : V} {p : G.Walk u v} (n : ℕ) (h : n < p.darts.length) :
+    p.darts[n].toProd = (p.getVert n, p.getVert (n + 1)) := by
```

Comment:
Can we have the version `p.darts[n] = _`, with a dart on the rhs (probably using `SimpleGraph.Walk.adj_getVert_succ`), and deduce this version from that?